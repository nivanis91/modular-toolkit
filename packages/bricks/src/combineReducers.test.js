import combineReducers from './combineReducers';

describe('When I combine some reducers', () => {
    let combinedReducer, state;
    beforeEach(() => {
        state = {
            baz: 'baz',
            argh: 'argh'
        };
        return (combinedReducer = combineReducers({
            baz: (s = 'baz', action) => {
                switch (action.type) {
                    case 'BAZ':
                        return 'BAZ!!!';
                    default:
                        return s;
                }
            },
            argh: (s = 'argh', action) => {
                switch (action.type) {
                    case 'ARGH':
                        return 'ARGH!!!';
                    default:
                        return s;
                }
            }
        }));
    });
    describe('and I call the combined reducer with an action for the first reducer', () => {
        let result;
        beforeEach(() => (result = combinedReducer(state, { type: 'BAZ' })));
        describe('the result', () => {
            it('is correct', () => {
                expect(result).toMatchSnapshot();
            });
        });
    });
    describe('and I call the combined reducer with an action for the second reducer', () => {
        let result;
        beforeEach(() => (result = combinedReducer(state, { type: 'ARGH' })));
        describe('the result', () => {
            it('is correct', () => {
                expect(result).toMatchSnapshot();
            });
        });
    });
});
