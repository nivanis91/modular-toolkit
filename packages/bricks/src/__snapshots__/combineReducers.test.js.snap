// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I combine some reducers and I call the combined reducer with an action for the first reducer the result is correct 1`] = `
Object {
  "argh": "argh",
  "baz": "BAZ!!!",
}
`;

exports[`When I combine some reducers and I call the combined reducer with an action for the second reducer the result is correct 1`] = `
Object {
  "argh": "ARGH!!!",
  "baz": "baz",
}
`;
